### PR TITLE
Fix doxygen warnings

### PR DESCRIPTION
**Pull request by leonid-ziskel**
Relates-to: OLPEDGE-731

Signed-off-by: Leonid Ziskel <leonid.ziskel@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates3d.h [hunk 1]
```diff
@@ -178,9 +178,9 @@ class CORE_API GeoCoordinates3d {
   bool IsValid() const;
 
  protected:
-  GeoCoordinates geo_coordinates_;
-  double altitude_;
-  static const double kNaN_;
+  GeoCoordinates geo_coordinates_;/*!< 2D geodetic coordinates */
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates3d.h]:
```suggestion
  /// 2D geodetic coordinates
  GeoCoordinates geo_coordinates_;
```

This is what we used elsewhere.

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoPoint.h [hunk 2]
```diff
@@ -41,17 +41,37 @@ class CORE_API GeoPoint {
...
    * Value range for each component is 32 bit unsigned integer
    */
   std::uint32_t x;
+  /**
+   * Absolute world Y-coordinate.
+   * Value range for each component is 32 bit unsigned integer
+   */
   std::uint32_t y;
-
+  /**
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoPoint.h]: Maybe we can use short-form `///` for the obvious methods (constr, destr, operators, etc.)
We dont need param and return.

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates3d.h (hunk 1 above)
Comment (leonid-ziskel) [reply olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates3d.h]: done

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoPoint.h (hunk 2 above)
Comment (leonid-ziskel) [reply olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoPoint.h]: done

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h [hunk 3]
```diff
@@ -56,8 +56,9 @@ class CORE_API GeoCoordinates {
    * [-90, 90] range.
    * @param longitude_degrees WGS84 longitude in degrees. Valid values are in
    * [-180, 180) range. Use normalized() to put geo coordinate in a valid range.
+   * @param degrees a struct to differ from the constructor with radians
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h]: Not needed. The official notion is *tag*, resp *tag dispatching*
https://www.boost.org/community/generic_programming.html#tag_dispatching
But i would omit it, except doxygen throws an error then maybe use following

```suggestion
   * @param Dispatch tag for degree coordinates.
```

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates3d.h [hunk 4]
```diff
@@ -178,8 +178,11 @@ class CORE_API GeoCoordinates3d {
   bool IsValid() const;
 
  protected:
+  /// 2D geodetic coordinates
   GeoCoordinates geo_coordinates_;
+  /// Altitude in meters
   double altitude_;
+  /// const being used to signalise an invalid altitude
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates3d.h]:
```suggestion
  /// Const being used to signalize an invalid altitude.
```

**Pull request merged**